app/services/allyship_creator.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+class AllyshipCreator
+  attr_reader :ally_id, :current_user, :notifier
+
+  def initialize(args)
+    @ally_id = args[:ally_id]
+    @current_user = args[:current_user]
+    @notifier = args[:notifier] || Allyships::AllianceNotifier
+  end
+
+  def self.perform(args = {})
+    new(args).perform
+  end
+
+  def perform
+    return unless valid?
+
+    setup_allyship
+    remove_request
+    notifier.perform(pusher_type: pusher_type,
+                     current_user: current_user,
+                     ally_id: ally_id)
+  end
+
+  private
+
+  def allyship
+    return @allyship if defined?(@allyship)
```

Comment:
Why not do `@allyship if @allyship`? (I haven't written Ruby in like 3 months lol)